nf_core/pipelines/lint/nf_test_content.py

```diff
@@ -0,0 +1,191 @@
+import re
+from pathlib import Path
+
+from nf_core.utils import load_tools_config
+
+
+def nf_test_content(self):
+    """Checks that the pipeline nf-test files have the appropriate content.
+
+    This lint test checks the following files and content of these files:
+
+    * `*.nf.test` files should specify the `outdir` parameter:
+
+        .. code-block:: yaml
+
+            when {
+                params {
+                    outdir = "$outputDir"
+                }
+            }
+
+    * A `versions.yml` file should be included in the snapshot of all `*.nf.test` files
+
+    * The `nextflow.config` file should contain:
+        .. code-block:: yaml
+            modules_testdata_base_path = <path>
+
+        .. code-block:: yaml
+            pipelines_testdata_base_path = <path>
+
+        And should set the correct resource limits, as defined in the `test` profile
+
+    * The `nf-test.config` file should:
+        * Make sure tests are relative to root directory
+
+            .. code-block:: yaml
+
+                testsDir "."
+
+        * Ensure a user-configurable nf-test directory
+
+            .. code-block:: yaml
+
+                workDir System.getenv("NFT_WORKDIR") ?: ".nf-test"
+
+        * Use a test specific config
+
+            .. code-block:: yaml
+
+                configFile "tests/nextflow.config"
+
+    All these checks can be skipped in the `.nf-core.yml` file using:
+
+        .. code-block:: yaml
+            lint:
+                nf_test_content: False
+
+        or
+
+        .. code-block:: yaml
+            lint:
+                nf_test_content:
+                    - outdir
+                    - versions.yml
+                    - modules_testdata_base_path
+                    - pipelines_testdata_base_path
+                    - resourceLimits
+                    - testsDir
+                    - workDir
+                    - configFile
```

Comment:
is it worth allowing to skip each one of the tests? or should we limit it to file names?
```

- <name>.nf.test

- nextflow.config

- nf-test.config
```

Reply:
changed to file names